docs/book/14-Streams.md

```diff
@@ -1138,7 +1138,7 @@ is it
 <!-- Optional -->
 ## Optional类
 
-在我们学习终端操作（Terminal Operations）之前，我们必须考虑如果你在一个空流中获取元素会发生什么。我们喜欢为了“happy path”而将流连接起来，并假设流不会被中断。在流中放置 `null` 是很好的中断方法。那么是否有某种对象，可作为流元素的持有者，即使查看的元素不存在也能友好地提示我们（也就是说，不会发生异常）？
+在我们学习终端操作（Terminal Operations）之前，我们必须考虑在一个空流中获取元素会发生什么。我们喜欢沿着“快乐路径”[^1]把流连接起来，同时假设流不会中断。然而，在流中放置 `null` 却会轻易令其中断。那么是否存在某种对象，可以在持有流元素的同时，即使在我们查找的元素不存在时，也能友好地对我们进行提示（也就是说，不会产生异常）？
```

Comment:
那么是否有某种对象，可作为流元素的持有者，即使查看的元素不存在也能友好地提示我们（也就是说，不会发生异常）？
后面这句话太拗口的。有点看不懂想干啥。

发个原文上来咱们一起看下吧

Reply:
这一段话的原文都放上来吧

Reply:
Before we can look at terminal operations, we must consider what
happens if you ask for an object in a stream and there's nothing
there. We like to connect up our streams for the "happy path" and
assume nothing will break. Putting a null in a stream is a good
way to break it. Is there some kind of object we can use that will
act as a holder for a stream element, but can also kindly tell us
(that is, no exceptions) if the element we're looking for isn't there?

This idea is implemented as the Optional type. Certain standard
stream operations return Optional objects because they cannot
guarantee the desired result will exist. These include: